library/src/main/java/com/novoda/downloadmanager/lib/DownloadExecutorFactory.java

```diff
@@ -1,24 +1,27 @@
 package com.novoda.downloadmanager.lib;
 
 import android.content.Context;
-import android.content.pm.ApplicationInfo;
-import android.content.pm.PackageManager;
-import android.os.Bundle;
 
-import com.novoda.notils.logger.simple.Log;
-
-import java.util.concurrent.ExecutorService;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
 class DownloadExecutorFactory {
+
     private static final int KEEP_ALIVE_TIME = 10;
-    private static final int DEFAULT_MAX_CONCURRENT_DOWNLOADS = 5;
-    private static final String METADATA_MAX_CONCURRENT_DOWNLOADS = "com.novoda.downloadmanager.MaxConcurrentDownloads";
 
-    public ExecutorService createExecutor(Context context) {
-        int maxConcurrentDownloads = getMaximumConcurrentDownloads(context);
+    private final ConcurrentDownloadsLimitProvider concurrentDownloadsLimitProvider;
+
+    static DownloadExecutorFactory newInstance(Context context) {
+        return new DownloadExecutorFactory(ConcurrentDownloadsLimitProvider.newInstance(context));
+    }
+
+    DownloadExecutorFactory(ConcurrentDownloadsLimitProvider concurrentDownloadsLimitProvider) {
+        this.concurrentDownloadsLimitProvider = concurrentDownloadsLimitProvider;
+    }
+
+    public ThreadPoolExecutor createExecutor() {
+        int maxConcurrentDownloads = concurrentDownloadsLimitProvider.getConcurrentDownloadsLimit();
```

Comment:
That's the idea behind the factory, it hides away the retrieval of that value so that the user of the executor doesn't have to know how to get the value.